custom_components/tuya_local/devices/airrobo_p20_vacuum.yaml

```diff
@@ -0,0 +1,289 @@
+    category: config
+    dps:
+      - id: 11
+        name: button
+        type: boolean
+        optional: true
+  - entity: button
+    name: Reset filter
+    class: restart
+    category: config
+    dps:
+      - id: 12
+        name: button
+        type: boolean
+        optional: true
+  - entity: sensor
+    name: Clean area
+    icon: "mdi:floor-plan"
+    category: diagnostic
+    dps:
+      - id: 16
+        type: integer
+        name: sensor
+        unit: m2
+        class: measurement
+  - entity: sensor
+    name: Clean time
+    class: duration
+    category: diagnostic
+    dps:
+      - id: 17
+        type: integer
+        name: sensor
+        unit: min
+        class: measurement
+  - entity: number
+    name: Volume
+    category: config
+    icon: "mdi:volume-high"
+    dps:
+      - id: 28
+        type: integer
+        name: value
+        unit: "%"
+        range:
+          min: 0
+          max: 100
+
+  - entity: binary_sensor
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 18
+        type: bitfield
+        name: sensor
+        optional: true
+        mapping:
+          - dps_val: 0
+            value: false
+          - value: true
+      - id: 18
+        type: bitfield
+        name: error
+        optional: true
+      - id: 18
+        type: bitfield
+        name: description
+        optional: true
+        mapping:
+          - dps_val: 0
+            value: OK
+          - dps_val: 1
+            value: Edge sweep
+          - dps_val: 2
+            value: Middle sweep
+          - dps_val: 4
+            value: Left wheel
+          - dps_val: 8
+            value: Right wheel
+          - dps_val: 16
+            value: Garage box
+          - dps_val: 32
+            value: Land check
+          - dps_val: 64
+            value: Collision
+          - dps_val: 128
+            value: Power low
+          - dps_val: 256
+            value: Raise up
+          - dps_val: 512
+            value: Safe place
+          - dps_val: 1024
+            value: Empty place
+          - dps_val: 2048
+            value: Fan error
+          - dps_val: 4096
+            value: Power lowsp
+          - dps_val: 8192
+            value: Middle sweep stop
+          - dps_val: 16384
+            value: Charging error
+          - dps_val: 32768
+            value: Charging no contact
+  - entity: select
+    name: Language
+    category: config
+    dps:
+      - id: 29
+        type: string
+        name: option
+        mapping:
+          - dps_val: english
+            value: English
+          - dps_val: german
+            value: German
+          - dps_val: french
+            value: French
+          - dps_val: russian
+            value: Russian
+          - dps_val: spanish
+            value: Spanish
+          - dps_val: italian
+            value: Italian
+  - entity: switch
+    name: Speed Mode
+    category: config
+    dps:
+      - id: 101
+        type: string

```

Comment:
need a `name: switch` in here.

Reply:
But I see this is duplicated with the select entity below, so one should be removed (I suspect you may have added the select to get around the switch not working without a dp named switch).